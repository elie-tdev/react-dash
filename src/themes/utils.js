export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
