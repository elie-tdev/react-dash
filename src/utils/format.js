const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
