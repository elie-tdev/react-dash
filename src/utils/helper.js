const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
