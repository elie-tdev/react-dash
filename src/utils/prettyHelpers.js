export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
