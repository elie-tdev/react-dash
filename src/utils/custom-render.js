import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
