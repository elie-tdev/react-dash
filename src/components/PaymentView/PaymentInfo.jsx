import { Box, Typography, makeStyles } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import cn from 'classnames';

import { formatDollar } from '@/utils/format';

const paymentInfoStyles = makeStyles(theme => ({
  paymentText: {
    color: props => (props.dark ? 'rgba(255, 255, 255, 0.5)' : 'rgba(50, 61, 88, 0.38)'),
  },
  value: {
    color: props => props.dark && theme.palette.primary.contrastText,
  },
  paymentTextContainer: {
    marginTop: '8px',
  },
}));
const PaymentInfo = ({
  paymentAmount,
  paymentDate,
  formattedDueDate = 'Jan 17, 2021',
  paymentType,
  dark,
}) => {
  const styles = paymentInfoStyles({ dark });
  const { t } = useTranslation();
  if (paymentType === 'paymentDue') {
    return (
      <Box data-testid="PaymentInfo" className={cn(styles.paymentTextContainer)}>
        <Typography className={cn(styles.paymentText)} variant="h5" display="inline">
          {t('content.Payment Due Body', {
            dueDate: formattedDueDate,
          })}
        </Typography>
      </Box>
    );
  }
  return (
    <Box data-testid="PaymentInfo" className={cn(styles.paymentTextContainer)}>
      <Typography className={cn(styles.paymentText)} variant="h5" display="inline">
        {t('content.Scheduled Payment Amount')}
      </Typography>
      <Typography className={cn(styles.value)} variant="h5" display="inline">
        {formatDollar(paymentAmount)}&nbsp;
      </Typography>
      <Typography className={cn(styles.paymentText)} variant="h5" display="inline">
        {paymentType === 'autoPay'
          ? t('content.Scheduled Date')
          : t('content.Scheduled Payment Charge Date')}
      </Typography>
      <Typography className={cn(styles.value)} variant="h5" display="inline">
        {paymentDate}
      </Typography>
      <Typography className={cn(styles.paymentText)} display="inline" variant="h5">
        .
      </Typography>
    </Box>
  );
};

export default PaymentInfo;
