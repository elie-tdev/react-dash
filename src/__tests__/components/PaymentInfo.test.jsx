import React from 'react';

import { render } from '@/utils/custom-render';
import PaymentInfo from '@/components/PaymentView/PaymentInfo';
test('Renders', () => {
  const { container } = render(<PaymentInfo paymentAmount="500" paymentDate="Jan 12, 2021" />);
  expect(container).toMatchSnapshot();
});
