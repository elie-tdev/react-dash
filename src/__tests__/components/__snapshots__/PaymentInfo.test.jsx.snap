// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6 makeStyles-paymentTextContainer-3"
    data-testid="PaymentInfo"
  >
    <h5
      class="MuiTypography-root makeStyles-paymentText-1 makeStyles-paymentText-4 MuiTypography-h5 MuiTypography-displayInline"
    >
      Your payment amount of 
    </h5>
    <h5
      class="MuiTypography-root makeStyles-value-2 makeStyles-value-5 MuiTypography-h5 MuiTypography-displayInline"
    >
      $500.00
      Â 
    </h5>
    <h5
      class="MuiTypography-root makeStyles-paymentText-1 makeStyles-paymentText-4 MuiTypography-h5 MuiTypography-displayInline"
    >
      will be charged on 
    </h5>
    <h5
      class="MuiTypography-root makeStyles-value-2 makeStyles-value-5 MuiTypography-h5 MuiTypography-displayInline"
    >
      Jan 12, 2021
    </h5>
    <h5
      class="MuiTypography-root makeStyles-paymentText-1 makeStyles-paymentText-4 MuiTypography-h5 MuiTypography-displayInline"
    >
      .
    </h5>
  </div>
</div>
`;
